### ISSUE TITLE
[12.0] Error A user cannot have both Tax B2B and Tax B2C, migrating account module. 

### ISSUE THREAD

**Issue by GSLabIt** [OCA/OpenUpgrade#2144]
Impacted versions: 12.0

Steps to reproduce:
1. in a v 11.0 db have Website Tax as B2C, so that Portal and/or Public Group will have B2C Tax.
2. start migrate to v 12.0 using `-u all`
3. error on account migration due to: 
https://github.com/OCA/OpenUpgrade/blob/bdf26ceb11834fd45ca4b4577532d6f5ee2ca4e4/addons/account/models/res_users.py#L24

Current behavior:
exception is raised because user has both B2B and B2C tax at same time. This seems to be caused by:
https://github.com/OCA/OpenUpgrade/blob/bdf26ceb11834fd45ca4b4577532d6f5ee2ca4e4/addons/account/migrations/12.0.1.1/noupdate_changes.xml#L108

as it force `account.group_show_line_subtotals_tax_excluded` to all users in groups Portal and Public.

Expected behavior: 
Do not force `account.group_show_line_subtotals_tax_excluded` in groups Portal and Public during migration.

Video/Screenshot link (optional):

**Comment by StefanRijnhart** [comment]
I see the problem. However, can you clarify "in a v 11.0 db have Website Tax as B2C" and how that leads to account.group_show_line_subtotals_tax_included being an implied group of Portal and Public given that this group is also introduced in Odoo 12?

**Comment by GSLabIt** [comment]
@StefanRijnhart will try to better explain the issue. 

On `website` settings, when u have `website_sale` installed, you can select if product will be shown as Tax Excluded or Tax Included:

<img width="1279" alt="Schermata 2020-01-03 alle 15 14 26" src="https://user-images.githubusercontent.com/11652845/71727815-bef69280-2e3b-11ea-9be9-1930f885b24e.png">

Setting option to `Tax Included` will change implied group for `Portal` and `Public` groups to `Tax Display B2C`, see following pics:

<img width="1270" alt="Schermata 2020-01-03 alle 15 16 45" src="https://user-images.githubusercontent.com/11652845/71727952-1d237580-2e3c-11ea-9bb1-39bb1f386559.png">
<img width="1272" alt="Schermata 2020-01-03 alle 15 16 25" src="https://user-images.githubusercontent.com/11652845/71727953-1d237580-2e3c-11ea-90c8-f4a584443a2d.png">

Now, migrating using OpenUpgrade will end to the error described above, by adding the 2 record in `no update_changes.xml`:
```
  <record id="base.group_portal" model="res.groups">
    <field name="implied_ids" eval="[(4, ref('account.group_show_line_subtotals_tax_excluded'))]"/>
  </record>
  <record id="base.group_public" model="res.groups">
    <field name="implied_ids" eval="[(4, ref('account.group_show_line_subtotals_tax_excluded'))]"/>
  </record>
```
Basically the 2 records will add `Technical Settings / Display Tax B2B` in both `Portal` and `Public` groups, so that the following ValidationError will be raised:

```
raise ValidationError(_("A user cannot have both Tax B2B and Tax B2C.\n"
                  "You should go in General Settings, and choose to display Product Prices\n"
                  "either in 'Tax-Included' or in 'Tax-Excluded' mode\n"
                  "(or switch twice the mode if you are already in the desired one)."))
```

Hope this will help to better understand the issue

**Comment by pedrobaeza** [comment]
I think it's reasonable to remove that noupdate data for preserving the behavior set on 11.0. They were not present in 11.0, but that's because the groups have been renamed (and handled in OU):

https://github.com/OCA/OpenUpgrade/blob/7d0aa1eb35faea71f44235d402a58f86789c5da4/addons/account/migrations/12.0.1.1/pre-migration.py#L52-L59

If we want to reset it, what we have to do is to put `(5, 0)` before in the noupdate data.

**Comment by StefanRijnhart** [comment]
Thanks for the clarification!

**Comment by pedrobaeza** [comment]
FYI I have discovered a bug in Odoo itself that triggers this error while no criteria problem was met:

https://github.com/odoo/odoo/pull/43964

**Comment by GSLabIt** [comment]
@pedrobaeza great one! I've approved PR.
